b2share/factory.py

```diff
@@ -1,192 +1,178 @@
+    app.logger.debug("Application created")
+
+@app_loaded.connect
+def receiver_app_loaded(sender, app=None, **kwargs):
+    app.logger.debug("Application Loaded")
+    app.logger.debug("Instance Path: {}".format(app.instance_path))
+    check_configuration(app.config, app.logger)
+
+def instance_path():
+    """Instance path for Invenio.
+
+    Defaults to ``<env_prefix>_INSTANCE_PATH``
+     or if environment variable is not set ``<sys.prefix>/var/instance``.
+    """
+    return os.getenv(env_prefix + '_INSTANCE_PATH') or \
+        os.path.join(sys.prefix, 'var', 'instance')
+
+
+def static_folder():
+    """Static folder path.
 
-instance_path = os.getenv(env_prefix + '_INSTANCE_PATH') or \
-    os.path.join(sys.prefix, 'var', 'b2share-instance')
-"""Instance path for B2Share."""
+    Defaults to ``<env_prefix>_STATIC_FOLDER``
+    or if environment variable is not set ``<sys.prefix>/var/instance/static``.
+    """
+    return os.getenv(env_prefix + '_STATIC_FOLDER') or \
+        os.path.join(instance_path(), 'static')
 
 
-def create_api(*args, **kwargs):
-    """Create Flask application providing B2SHARE REST API."""
-    app = create_app_factory(
-        'b2share',
-        config_loader=config_loader,
-        extension_entry_points=['invenio_base.api_apps'],
-        blueprint_entry_points=['invenio_base.api_blueprints'],
-        converter_entry_points=['invenio_base.api_converters'],
-        instance_path=instance_path,
-    )(*args, **kwargs)
-    return app
+def static_url_path():
+    """Static url path.
 
+    Defaults to ``<env_prefix>_STATIC_URL_PATH``
+    or if environment variable is not set ``/static``.
+    """
+    return os.getenv(env_prefix + '_STATIC_URL_PATH') or '/static'
 
-def create_app(**kwargs):
-    """Create Flask application providing B2SHARE UI and REST API.abs
 
-    The REST API is provided by redirecting any request to another Flask
-    application created with :func:`~.create_api`.
+def config_loader(app, **kwargs_config):
+    """Configuration loader.
+
+    Adds support for loading templates from the Flask application's instance
+    folder (``<instance_folder>/templates``).
     """
-    # Create the REST API Flask application
-    api = create_api(**kwargs)
-    api.config.update(
-        APPLICATION_ROOT='/api'
+    # This is the only place customize the Flask application right after
+    # it has been created, but before all extensions etc are loaded.
+    local_templates_path = os.path.join(app.instance_path, 'theme/templates')
+    if os.path.exists(local_templates_path):
+        # Let's customize the template loader to look into packages
+        # and application templates folders.
+        app.jinja_loader = ChoiceLoader([
+            FileSystemLoader(local_templates_path),
+            app.jinja_loader,
+        ])
+
+    app.jinja_options = dict(
+        app.jinja_options,
+        cache_size=1000,
+        bytecode_cache=BytecodeCache(app)
     )
```

Comment:
HK: Might not be used. Comes from cookiecutter. B2SHARE V2 has a very simple config_loader: https://github.com/EUDAT-B2SHARE/b2share/blob/d846068c71853221ac53239c9efc56e3e795e6c3/b2share/factory.py#L101

Reply:
**Let's check if this can be removed?**